app/services/discovery_report.rb

```diff
@@ -38,7 +38,7 @@ def each_row
       row.counts[:mimetypes].each { |k, v| summary[:mimetypes][k] += v }
       # log the output to a running progress file
       File.open(batch.batch_context.progress_log_file, 'a') { |f| f.puts log_progress_info(dobj, status).to_yaml }
-      yield row
+      yield row.to_h
```

Comment:
the hash is the only thing that was being used by the yielded result (for JSON serialization), and returning just the hash instead of the whole validator object made an important performance difference (see commit msg).